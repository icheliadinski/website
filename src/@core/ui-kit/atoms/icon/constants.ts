export enum ICON_TYPES {
  PHONE = 'phone',
  GLASSES = 'glasses',
  USER = 'user',
  MONITOR = 'monitor',
  EDIT = 'edit',
  DOWNLOAD = 'download',
  WORK = 'work',
  ROCKET = 'inclined-rocket',
  DATABASE = 'database',
  COMPUTER = 'computer',
  ITERATION = 'iteration',
  ENVELOPE = 'envelope',
  ADDRESS = 'address',
  SPEEDOMETER = 'speedometer',
  MORTARBOARD = 'mortarboard',
  CONTRACT = 'contract',
  FACEBOOK = 'facebook-logo',
  GITHUB = 'github-logo',
  TWITTER = 'twitter',
}
