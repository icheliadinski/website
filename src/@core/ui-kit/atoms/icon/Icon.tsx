import React from 'react'
import classNames from 'classnames'
import { ICON_TYPES } from './constants'

export interface Props {
  type: ICON_TYPES
}

export const Icon: React.FC<Props> = ({ type }) => {
  const classes = classNames('icon', `icon-${type}`)
  return <span className={classes} />
}
