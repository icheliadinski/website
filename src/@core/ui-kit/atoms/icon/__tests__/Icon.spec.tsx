import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { Icon, Props } from '../Icon'
import { ICON_TYPES } from '../constants'

describe('[Icon]', () => {
  let wrapper: ShallowWrapper<Props>

  it('renders successfully with download icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.DOWNLOAD} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with edit icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.EDIT} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with glasses icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.GLASSES} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with monitor icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.MONITOR} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with phone icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.PHONE} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.USER} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.USER} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.WORK} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.ROCKET} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.DATABASE} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.COMPUTER} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.ITERATION} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.ENVELOPE} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.ADDRESS} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.SPEEDOMETER} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.MORTARBOARD} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.CONTRACT} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.FACEBOOK} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.GITHUB} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders successfully with user icon', () => {
    wrapper = shallow(<Icon type={ICON_TYPES.TWITTER} />)
    expect(wrapper).toMatchSnapshot()
  })
})
