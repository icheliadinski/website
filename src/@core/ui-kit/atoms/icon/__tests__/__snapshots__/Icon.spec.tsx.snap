// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Icon] renders successfully with download icon 1`] = `
<span
  className="icon icon-download"
/>
`;

exports[`[Icon] renders successfully with edit icon 1`] = `
<span
  className="icon icon-edit"
/>
`;

exports[`[Icon] renders successfully with glasses icon 1`] = `
<span
  className="icon icon-glasses"
/>
`;

exports[`[Icon] renders successfully with monitor icon 1`] = `
<span
  className="icon icon-monitor"
/>
`;

exports[`[Icon] renders successfully with phone icon 1`] = `
<span
  className="icon icon-phone"
/>
`;

exports[`[Icon] renders successfully with user icon 1`] = `
<span
  className="icon icon-user"
/>
`;

exports[`[Icon] renders successfully with user icon 2`] = `
<span
  className="icon icon-user"
/>
`;

exports[`[Icon] renders successfully with user icon 3`] = `
<span
  className="icon icon-work"
/>
`;

exports[`[Icon] renders successfully with user icon 4`] = `
<span
  className="icon icon-inclined-rocket"
/>
`;

exports[`[Icon] renders successfully with user icon 5`] = `
<span
  className="icon icon-database"
/>
`;

exports[`[Icon] renders successfully with user icon 6`] = `
<span
  className="icon icon-computer"
/>
`;

exports[`[Icon] renders successfully with user icon 7`] = `
<span
  className="icon icon-iteration"
/>
`;

exports[`[Icon] renders successfully with user icon 8`] = `
<span
  className="icon icon-envelope"
/>
`;

exports[`[Icon] renders successfully with user icon 9`] = `
<span
  className="icon icon-address"
/>
`;

exports[`[Icon] renders successfully with user icon 10`] = `
<span
  className="icon icon-speedometer"
/>
`;

exports[`[Icon] renders successfully with user icon 11`] = `
<span
  className="icon icon-mortarboard"
/>
`;

exports[`[Icon] renders successfully with user icon 12`] = `
<span
  className="icon icon-contract"
/>
`;

exports[`[Icon] renders successfully with user icon 13`] = `
<span
  className="icon icon-facebook-logo"
/>
`;

exports[`[Icon] renders successfully with user icon 14`] = `
<span
  className="icon icon-github-logo"
/>
`;

exports[`[Icon] renders successfully with user icon 15`] = `
<span
  className="icon icon-twitter"
/>
`;
